File: Python/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        tmp = []
        dict = {")" : "(", "}" : "{", "]" : "["}
        for c in s:
            if c in dict.values():
                tmp.append(c)
            elif c in dict.keys():
                if tmp == [] or dict[c] != tmp.pop():
                    return False
            else:
                return False
        return tmp == []

File: Java/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> tmp = new Stack<Character>();
        for(char c : s.toCharArray()){
            if( c == '(' || c == '{' || c == '['){
                tmp.push(c);
            }
            else{
                if((tmp.empty()) || ( c == ')' && tmp.peek() != '(') || ( c == '}' && tmp.peek() != '{') || ( c == ']' && tmp.peek() != '['))
                    return false;
                tmp.pop();
            }
        }
        return tmp.empty();
    }
}